lib/RGBot.js

```diff
@@ -583,196 +1098,360 @@ const RGBot = class {
      * @param {number} [options.maxDistance=50] Find any Items matching the search criteria up to and including this distance from the Bot
      * @returns {Item[]} The list of Items found on the ground (can be empty)
+     *
+     * @deprecated Since 1.2.0, please use findItems(options, itemValueFunction, sortValueFunction)
      */
     findItemsOnGround(options = {}) {
-        const itemName = options.itemName || undefined;
-        const partialMatch = options.partialMatch || false;
+        this.#logWarn(`findItemsOnGround(options): Deprecated since 1.2.0, please use findItems(options, itemValueFunction, sortValueFunction)`)
+        options['itemNames'] = options.itemName && [options.itemName] || [];
+        return this.findItemsOnGroundv2(options).shift()?.result
+    }
+
+    /**
+     * Returns a list of all Items that are on the ground within a maximum distance from the Bot (can be empty).
+     * @param {object} [options] - optional parameters
+     * @param {string[]} [options.itemNames=[]] Find only Items matching one of these names
+     * @param {boolean} [options.partialMatch=false] If itemName is defined, find Items whose names / displayNames contain itemName. (Ex. '_boots' may find any of 'iron_boots', 'golden_boots', etc.)
+     * @param {number} [options.maxDistance=50] - find any Items matching the search criteria up to and including this distance from the Bot
+     * @param {number} [options.maxCount=1] - limit the number of items to find
+     * @param {function(blockName:string):number} [itemValueFunction] - function to call to get the value of an item based on its name (itemName); should return the best item with the lowest sorting value. A good example function is { return scoreValueOf[itemName] }, where scoreValueOf is the point value or intrinsic value of the item in the game mode being played.  If you don't want an item considered, return a value < 0 for its value.
+     * @param {function(pointValue:number,distance:number):number} [sortValueFunction] - function to call to sort the evaluation of results based on arguments (pointValue, distance); should return the best item with the lowest sorting value.  A good example function is {return -1 * (pointValue*2 - distance*1.1)}
+     *
+     * @returns {FindResult[]} - the best items found
+     *
+     * By default, this only sorts results by distance, and assigns all items a pointValue of 0, but a good example function for considering pointValue would be {return -1 * (pointValue*2 - distance*1.1)) }
+     *
+     * To get the 'best' item to collect, call findItems(...).shift().  Note that the result may be null|undefined if no items were found
+     */
+    findItemsOnGroundv2(options = {}, itemValueFunction = (itemName) => {return 0}, sortValueFunction = (pointValue,distance) => {return distance}) {
```

Comment:
Similar comment re parameters and options.

Other comment - why is this named v2? Why are we not re-using the old name? It seems to have the same signature, once we move the optionals into the options object.

Reply:
return type is different :(

This one pains me greatly... I would love to use the old name.

Really we need to decide
1. Do we deprecate any of these, or just leave the simple vs advanced both in there (maybe we label them that way in the JSDoc)
2. For APIs like this, can we just nuke the old one given where we are in the project.

Reply:
Having both simple and advanced ones would be ideal in my opinion... like in this case, we can probably name it `findBestItemsOnGround`?

Reply:
I'm going with (remove the old method since we don't have active users yet).

I will update the bot sample templates if needed

Reply:
I was going to say replace the old method anyway if it's just due to the return types... because we talked about changing the return types for all of these at some point such that you'd get back more info about the action like how long it took to complete